components/CostsLegend.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <ChartLegend
+    :items="items"
+    :row-height-rem="0.7"
+  />
+</template>
+
+<script setup lang="ts">
+import type { ScenarioCost } from "~/types/resultTypes";
+import { costsPieColors, type LegendItem, LegendShape } from "./utils/charts";
+
+const appStore = useAppStore();
+
+const costLabel = (cost: ScenarioCost) => {
+  return appStore.metadata?.results.costs.find(costMeta => costMeta.id === cost.id)?.label;
+};
+
+const items = computed((): LegendItem[] => {
+  const costsWithColors = appStore.totalCost?.children?.map((cost: ScenarioCost, index: number) => {
+    return { color: costsPieColors[index + 1], label: costLabel(cost), shape: LegendShape.Rectangle, value: cost.value };
```

Comment:
index + 1 because total gets color 0 and we don't include that in the legend?

Reply:
Well inferred, I can add a comment